### PR TITLE
scans/stack: add slicing by timestamp and string

### PR DESCRIPTION
**Pull request by JoepVanlier**
**Why this PR?**
It's very convenient to slice by time and timestamp, hence we want to extend this functionality to `Scan` and `Stack`.

### CODE REVIEW

File: lumicks/pylake/detail/imaging_mixins.py [hunk 1]
```diff
@@ -72,3 +77,30 @@ def plot(num):
         )
         line_ani.save(file_name, writer=writer)
         plt.close(fig)
+
+
+class FrameIndex:
+    def to_frame_index(self, time, use_start=False) -> Union[int, None]:
+        """Convert timestamps, timestamps or frame indices to frame indices
```
Comment (rpauszek) [inline lumicks/pylake/detail/imaging_mixins.py:84]: rebasing mishap?
```suggestion
        """Convert timestamps to frame indices
```

File: lumicks/pylake/detail/imaging_mixins.py [hunk 2]
```diff
@@ -72,3 +77,30 @@ def plot(num):
...
+
+        Parameters
+        ----------
+        time : str or int or None
+            Time expressed in either string format (e.g. "5s"), timestamp in nanoseconds or frame.
+        use_start : bool
+            Are we dealing with the start or end of a frame here?
```
Comment (rpauszek) [inline lumicks/pylake/detail/imaging_mixins.py:91]: The term "use" here seems odd, I wasn't quite sure what the parameter meant. I think "is" might be better and more in line with convention for naming boolean args.

The description seems a bit off also; until I looked at the usage in the code I was thinking this arg meant something completely different. Not sure if I quite got the intended usage, but how about:
```suggestion
        is_frame_start : bool
            Whether the input time refers to the start or stop timestamp of the searched frame
```

File: lumicks/pylake/detail/imaging_mixins.py [hunk 3]
```diff
@@ -72,3 +77,30 @@ def plot(num):
         )
         line_ani.save(file_name, writer=writer)
         plt.close(fig)
+
+
+class FrameIndex:
+    def to_frame_index(self, time, use_start=False) -> Union[int, None]:
```
Comment (rpauszek) [inline lumicks/pylake/detail/imaging_mixins.py:83]: should be private API?

also perhaps `_time_to_frame_index` might be more descriptive?

**Review by rpauszek — CHANGES_REQUESTED** [review]
Looks pretty good to me. Just some concerns with the descriptions/docstrings I found confusing and future me will likely be completely lost, but let's see if you agree.

The big thing is I think the new method introduced through the mixin should be private API

File: docs/tutorial/correlatedstacks.rst [hunk 4]
```diff
@@ -11,6 +11,10 @@ These videos can be opened and sliced using `CorrelatedStack`::
     stack = lk.CorrelatedStack("wf.tiff")  # Loading a stack.
     stack_slice = stack[2:10]  # Grab frame 2 to 9
 
+You can also slice by time::
+
+    stack_slice = stack["2s":"10s"]  # Slice from second second up to tenth second
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply docs/tutorial/correlatedstacks.rst:16]: second second 😅

File: docs/tutorial/images.rst [hunk 5]
```diff
@@ -55,6 +55,14 @@ This will return a new `Scan` containing data equivalent to::
 
     multiframe_scan.get_image("rgb")[5:10, :, :, :]
 
+We can also slice the frames by time::
+
+    sliced_scan = multiframe_scan["5s":"10s"]
```
[the remark this answers is not in the document]
Comment (JoepVanlier) [reply docs/tutorial/images.rst:60]: How would you feel about `# get frames corresponding to the time range 5 through 10 seconds`?

File: lumicks/pylake/detail/imaging_mixins.py (hunk 2 above)
Comment (JoepVanlier) [reply lumicks/pylake/detail/imaging_mixins.py:91]: I agree with your suggestion. I think your names and docstring description are better here.

File: lumicks/pylake/detail/imaging_mixins.py (hunk 3 above)
Comment (JoepVanlier) [reply lumicks/pylake/detail/imaging_mixins.py]: Yes, should definitely be private API. Good catch!

File: docs/tutorial/images.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (rpauszek) [reply docs/tutorial/images.rst:61]: yup that sounds good

**Review by rpauszek — APPROVED** [review]
cool, I think it's good to go

**Pull request merged**